core/src/main/scala/kafka/utils/Implicits.scala

```diff
@@ -46,4 +47,21 @@ object Implicits {
 
   }
 
+  /**
+   * Exposes `foreachKv` which maps to `foreachEntry` in Scala 2.13 and `foreach` in Scala 2.12
+   * (with the help of scala.collection.compat). `foreachEntry` avoids the tuple allocation and
+   * is more efficient.
+   *
+   * This was not named `foreachEntry` to avoid `unused import` warnings in Scala 2.13 (the implicit
+   * would not be triggered in Scala 2.13 since `Map.foreachEntry` would have precedence).
+   */
+  @nowarn("cat=unused-imports")
+  implicit class MapExtensionMethods[K, V](private val self: scala.collection.Map[K, V]) extends AnyVal {
+    import scala.collection.compat._
+    def foreachKv[U](f: (K, V) => U): Unit = {
```

Comment:
The name reads a tad awkwardly. I wonder if `foreachKeyValue` would be too verbose. Or maybe `foreachMapEntry`?

Reply:
Or `foreachPair`.

Reply:
Thanks for the feedback. I agree that the name could be improved. I was hoping for something concise and descriptive. `foreachPair` is not bad, but you'd expect to get a pair (tuple) and this is to avoid that. How about `forKeyValue`?

Reply:
Also, note that this will go away once we drop support for Scala 2.12.

Reply:
`forKeyValue` LGTM since `for` is the language's keyword for `foreach` or `map`.

Reply:
That was the motivation indeed. Great, I'll change it.

Reply:
Checked with @hachikuji and he's fine with this change.